### PR TITLE
Br 2807

### PR DESCRIPTION
**Pull request by dhiraj-toptal**

### CODE REVIEW

File: Model/RefundCanceledProcessor.php [hunk 1]
```diff
@@ -66,12 +79,9 @@ public function processCanceledWebhook($params, $jobName)
...
             ->getFirstItem()->getCreditMemoId();
         if ($memoId) {
-            $creditmemoModel = $this->creditmemoFactory->create();
-            $creditmemoModel->setData([
-                'entity_id' => $memoId,
-                'creditmemo_status' => 1
-            ]);
-            $creditmemoModel->save();
+            $memoData = $this->creditmemoRepository->get($memoId);
+            $memoData->addData(['state' => 1]);
```
Comment (ynaxonBalance) [inline Model/RefundCanceledProcessor.php]: can we represent state in textual value?

File: Model/RefundFailedProcessor.php [hunk 2]
```diff
@@ -56,12 +72,9 @@ public function processFailedWebhook($params, $jobName)
...
             ->getFirstItem()->getCreditMemoId();
         if ($memoId) {
-            $creditmemoModel = $this->creditmemoFactory->create();
-            $creditmemoModel->setData([
-                'entity_id' => $memoId,
-                'creditmemo_status' => 1
-            ]);
-            $creditmemoModel->save();
+            $memoData = $this->creditmemoRepository->get($memoId);
+            $memoData->addData(['state' => 1]);
```
Comment (ynaxonBalance) [inline Model/RefundFailedProcessor.php]: can we represent state in textual value?

File: Model/RefundSuccessfulProcessor.php [hunk 3]
```diff
@@ -47,21 +49,18 @@ public function __construct(
...
             ->getFirstItem()->getCreditMemoId();
         if ($memoId) {
-            $creditmemoModel = $this->creditmemoFactory->create();
-            $creditmemoModel->setData([
-                'entity_id' => $memoId,
-                'creditmemo_status' => 2
-            ]);
-            $creditmemoModel->save();
+            $memoData = $this->creditmemoRepository->get($memoId);
+            $memoData->addData(['state' => 2]);
```
Comment (ynaxonBalance) [inline Model/RefundSuccessfulProcessor.php]: can we represent state in textual value?

File: Observer/SalesOrderCreditmemoSaveAfter.php [hunk 4]
```diff
@@ -61,8 +61,19 @@ public function execute(Observer $observer)
...
+            ->addFieldToFilter('credit_memo_id', ['eq' => $creditMemoId])->getData();
+        if (count($balancepayRefundData)>0) {
+            return;
+        }
         $total = $creditMemo->getBaseGrandTotal();
         $invoiceId = $creditMemo->getInvoiceId();
+        $comments = $creditMemo->getComments();
+        $reason = 'Other';
+        foreach ($comments as $comment) {
+            $reason = $comment->getComment();
```
Comment (ynaxonBalance) [inline Observer/SalesOrderCreditmemoSaveAfter.php:74]: what is comments?

Comment (dhiraj-toptal) [reply Observer/SalesOrderCreditmemoSaveAfter.php:74]: Comments is the reason value from Dropdown

Comment (ynaxonBalance) [reply Observer/SalesOrderCreditmemoSaveAfter.php:74]: why not reason?

Comment (dhiraj-toptal) [reply Observer/SalesOrderCreditmemoSaveAfter.php:74]: We have replace the textbox for comments with Reason Dropdown so had to use the comments input here.

File: Model/RefundSuccessfulProcessor.php (hunk 3 above)
Comment (dhiraj-toptal) [reply Model/RefundSuccessfulProcessor.php]: Updated

File: Model/RefundFailedProcessor.php (hunk 2 above)
Comment (dhiraj-toptal) [reply Model/RefundFailedProcessor.php]: Updated

File: Model/RefundCanceledProcessor.php (hunk 1 above)
Comment (dhiraj-toptal) [reply Model/RefundCanceledProcessor.php]: Updated

**Review by ynaxonBalance — APPROVED** [review]

**Pull request merged**